DQM/BeamMonitor/plugins/OnlineBeamMonitor.cc

```diff
@@ -0,0 +1,398 @@
+/*
+ * \file OnlineBeamMonitor.cc
+ * \author Lorenzo Uplegger/FNAL
+ * modified by Simone Gennai INFN/Bicocca
+ */
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DQM/BeamMonitor/plugins/OnlineBeamMonitor.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/View.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "RecoVertex/BeamSpotProducer/interface/BeamFitter.h"
+#include "RecoVertex/BeamSpotProducer/interface/PVFitter.h"
+#include <memory>
+
+#include <numeric>
+
+using namespace std;
+using namespace edm;
+using namespace reco;
+
+//----------------------------------------------------------------------------------------------------------------------
+OnlineBeamMonitor::OnlineBeamMonitor(const ParameterSet& ps)
+    : monitorName_(ps.getUntrackedParameter<string>("MonitorName")),
+      bsTransientToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()),
+      bsHLTToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()),
+      bsLegacyToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()),
+      numberOfValuesToSave_(0) {
+  if (!monitorName_.empty())
+    monitorName_ = monitorName_ + "/";
+
+  processedLumis_.clear();
+
+  varNamesV_.push_back("x");
+  varNamesV_.push_back("y");
+  varNamesV_.push_back("z");
+  varNamesV_.push_back("sigmaX");
+  varNamesV_.push_back("sigmaY");
+  varNamesV_.push_back("sigmaZ");
+
+  //histoByCategoryNames_.insert(pair<string, string>("run", "Coordinate"));
+  //histoByCategoryNames_.insert(pair<string, string>("run", "PrimaryVertex fit-DataBase"));
+  //histoByCategoryNames_.insert(pair<string, string>("run", "PrimaryVertex fit-BeamFit"));
+  //histoByCategoryNames_.insert(pair<string, string>("run", "PrimaryVertex fit-Scalers"));
+  //histoByCategoryNames_.insert(pair<string, string>("run", "PrimaryVertex-DataBase"));
+  //histoByCategoryNames_.insert(pair<string, string>("run", "PrimaryVertex-BeamFit"));
+  //histoByCategoryNames_.insert(pair<string, string>("run", "PrimaryVertex-Scalers"));
+
+  histoByCategoryNames_.insert(pair<string, string>("lumi", "Lumibased BeamSpotHLT"));
+  histoByCategoryNames_.insert(pair<string, string>("lumi", "Lumibased BeamSpotLegacy"));
+  histoByCategoryNames_.insert(pair<string, string>("lumi", "Lumibased BeamSpotTransient"));
+
+  for (vector<string>::iterator itV = varNamesV_.begin(); itV != varNamesV_.end(); itV++) {
```

Comment:
Please move to the C++11 range-based loop. Please use `const` (unless you want to modify the iterator). For this line it should be:
```suggestion
  for (const auto& itV :  varNamesV_) {
```

Reply:
done. Will make a commit as soon as I finish implementing the rest of the comments

Reply:
I think I need also to modify the following line:
histosMap_[*itV][itM->first][itM->second] = nullptr;
to
histosMap_[itV][itM->first][itM->second] = nullptr;

otherwise it does not compile.